File: backend/app/schemas/requester.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RequesterBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    full_name: str
    requester_type: str
    affiliation: str | None = None
    region: str | None = None
    category: str
    is_repeat_complainant: bool


class RequesterOut(RequesterBrief):
    total_appeals: int
    resolved_appeals: int
    rejected_appeals: int
    first_appeal_date: datetime | None = None
    last_appeal_date: datetime | None = None
    category_score: float
    repeat_score: float
    top_topics: list
    top_regions: list
    behavior_stats: dict
    created_at: datetime


class PaginatedRequesters(BaseModel):
    items: list[RequesterOut]
    total: int
    page: int
    page_size: int

File: frontend/src/types/requester.ts
export interface RequesterBrief {
  id: number
  full_name: string
  requester_type: string
  affiliation?: string | null
  region?: string | null
  category: string
  is_repeat_complainant: boolean
}

export interface Requester extends RequesterBrief {
  total_appeals: number
  resolved_appeals: number
  rejected_appeals: number
  first_appeal_date?: string | null
  last_appeal_date?: string | null
  category_score: number
  repeat_score: number
  top_topics: string[]
  top_regions: string[]
  behavior_stats: { avg_per_month?: number; top_topic_share?: number }
  created_at: string
}
